src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+
+        result.append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (final String name : names) {
+            result.append(System.lineSeparator());
+            int salary = 0;
+
+            for (final String datum : data) {
+                String[] datumSplit = datum.split(" ");
+                LocalDate current = LocalDate.parse(datumSplit[0], formatter);
+
+                if (name.equals(datumSplit[1])) {
```

Comment:
constants for indexes

Reply:
added them to the method.
I can't use only UPPERCASE letters, because check style plugin shows errors and requires to use camelCase letters